### PR TITLE
Text Analytics v3.2 preview.1 

### PR DESCRIPTION
**Pull request by abhahn**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by abhahn** [comment]
Hi @yangyuan , the build for this PR is currently breaking on a cross-version breaking change check.  This new preview version is actually based on the last stable version (v3.1), but is being compared to our last preview version (v3.1-preview.5).  Is there some way I can force the build to compare with the latest stable version instead, in which case there shouldn't be any breaking changes?

**Comment by jianyexi** [comment]
The cross version breaking change can be ignored, as it compares with the last preview version , per the result compared with the last stable version , there is no breaking change,
https://portal.azure-devex-tools.com/tools/diff/eyJzb3VyY2VVcmwiOiJodHRwczovL2dpdGh1Yi5jb20vQXp1cmUvYXp1cmUtcmVzdC1hcGktc3BlY3MvYmxvYi9tYXN0ZXIvc3BlY2lmaWNhdGlvbi9jb2duaXRpdmVzZXJ2aWNlcy9kYXRhLXBsYW5lL1RleHRBbmFseXRpY3Mvc3RhYmxlL3YzLjEvVGV4dEFuYWx5dGljcy5qc29uIiwidGFyZ2V0VXJsIjoiaHR0cHM6Ly9naXRodWIuY29tL0F6dXJlL2F6dXJlLXJlc3QtYXBpLXNwZWNzL2Jsb2IvZDc1OGM0MjA1ZDMzMWM1NTJjYWZiYjc1NWVkMDI2NzNiOWZhNWUyMi9zcGVjaWZpY2F0aW9uL2NvZ25pdGl2ZXNlcnZpY2VzL2RhdGEtcGxhbmUvVGV4dEFuYWx5dGljcy9wcmV2aWV3L3YzLjItcHJldmlldy4xL1RleHRBbmFseXRpY3MuanNvbiJ9

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json [hunk 1]
```diff
@@ -0,0 +1,3118 @@
...
+            },
+            "sentenceCount": {
+              "default": 3,
+              "type": "integer"
+            },
+            "sortBy": {
+              "default": "Offset",
+              "enum": [
+                "Offset",
+                "Rank"
```
Comment (deyaaeldeen) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:824]: are we sure capitalization is correct for these values? that burned me in testing so many times lol

Comment (abhahn) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:824]: Since this is represented as an enum in C#, and the values are passed in the body rather than query params (as in our sync services), then I think these values should be capitalized.  I was running into issues without capitalization when testing with Analyze, so that's why I changed this.  Is this consistent with other parameter values for other Analyze tasks?

Comment (mssfang) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:824]: Capitalization of first letter of enum values is correct.

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json [hunk 2]
```diff
@@ -0,0 +1,3118 @@
...
+          "description": "Warning message."
+        },
+        "targetRef": {
+          "type": "string",
+          "description": "A JSON pointer reference indicating the target object."
+        }
+      }
+    },
+    "ExtractiveSummarizationTask": {
+      "properties": {
```
Comment (kristapratico) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:802]:
```suggestion
    "ExtractiveSummarizationTask": {
      "type": "object",
      "properties": {
```

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json [hunk 3]
```diff
@@ -0,0 +1,3118 @@
...
+            },
+            "sortBy": {
+              "default": "Offset",
+              "enum": [
+                "Offset",
+                "Rank"
+              ],
+              "type": "string"
+            }
+          }
```
Comment (kristapratico) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:828]: should `ExtractiveSummarizationTask` also have a `taskName` property?

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json [hunk 4]
```diff
@@ -0,0 +1,3118 @@
...
+        }
+      }
```
Comment (mssfang) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:830]:
```suggestion
          },
          "type": "object"
        },	
        "taskName": {	
          "type": "string"	
        }	
      },	
      "type": "object"
      }
```

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json (hunk 2 above)
Comment (mssfang) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:802]: better align with existing ordering: https://github.com/Azure/azure-rest-api-specs/pull/15159/files#r670782840

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json (hunk 1 above)
Comment (deyaaeldeen) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:824]: I confirm it works correctly in PPE, thanks!

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json [hunk 5]
```diff
@@ -0,0 +1,3118 @@
...
+              "type": "boolean"
+            },
+            "stringIndexType": {
+              "$ref": "#/definitions/StringIndexType"
+            },
+            "sentenceCount": {
+              "default": 3,
+              "type": "integer"
+            },
+            "sortBy": {
```
Comment (maririos) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:820]: looking at the REST API guidelines, the recommendation is to use `orderBy` instead of `sortBy`. @abhahn do you know if this could be renamed?

https://github.com/microsoft/api-guidelines/blob/azureRestUpdates/azure/Guidelines.md#query-options

Comment (maririos) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:820]: although this applies to collections, I wonder if it will be ok to unify,

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json [hunk 6]
```diff
@@ -0,0 +1,3118 @@
...
+    "ExtractiveSummarizationTask": {
+      "properties": {
+        "parameters": {
+          "properties": {
+            "model-version": {
+              "default": "latest",
+              "type": "string"
+            },
+            "loggingOptOut": {
+              "default": true,
```
Comment (kristapratico) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:810]: @abhahn should the `loggingOptOut` default to `false` here?

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json (hunk 5 above)
Comment (abhahn) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:820]: I think at this point it is too late to change for this preview version, but I can include this as a change to include in the next preview version to align with the guidelines.  I'm surprised this didn't come up during the API Review back in May.

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json (hunk 2 above)
Comment (abhahn) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json]: I have added `type: "object"` here, but it looks like the ordering is not consistent throughout the Swagger when looking at other objects.  Due to time constraints and having other higher priority stuff to work on at the moment I don't consider the ordering to be a high priority change to make throughout, but if it is important I can create a work item to fix after the release.

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json (hunk 3 above)
Comment (abhahn) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:829]: Yep, thanks for catching!  I've added this now.

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json (hunk 4 above)
Comment (abhahn) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json]: I've just added the taskName and the object types.

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json (hunk 6 above)
Comment (abhahn) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json]: Yep, updated!

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json (hunk 5 above)
Comment (maririos) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:822]: For sure. It could def happen in another version.
It could have been missed or because it is not a pageable sort? Idk...

File: specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json [hunk 7]
```diff
@@ -846,7 +850,7 @@
               "type": "string"
             },
             "loggingOptOut": {
-              "default": true,
+              "default": false,
```
Comment (kristapratico) [inline specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:854]: sorry to bug you on this again, but was this change intentional for `PiiTask`? I thought we wanted to opt out of logging specifically for PII and healthcare, but correct me if I'm wrong about this.

Comment (abhahn) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json:854]: Oh ok, yes, this was probably unintentional.  I might have been looking at the wrong line of code when I did this.  Good to know that logging opt out is intentionally different for those two services.  I'll double check things and check in a fix soon.

Comment (abhahn) [reply specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.2-preview.1/TextAnalytics.json]: Reverted this change.  Let me know if you notice any other issues.  :)

**Review by kristapratico — APPROVED** [review]

**Review by deyaaeldeen — APPROVED** [review]

**Review by mssfang — APPROVED** [review]
